### PR TITLE
b2b gemm residual

### PR DESCRIPTION
**Pull request by hwu36**
Add residual support for shmem staging iterator used in back-to-back GEMM fusion. This allows support of problem_size_0_n that is not multiple of 32.

@danthe3rd , would you please give it a try?

### CODE REVIEW

**Comment by danthe3rd** [comment]
Looking quickly through the code it looks great! Thanks a lot for putting that up so fast :)
Unfortunately, I'll be away for the entire month of August. I can give it a try in September, I'll also try to find someone else to do it in my absence - but in the meantime feel free to merge. If it passes the tests for values of `problemSize1.K` like `[16, 32, 34, 36, 48, 64, 80]`, it should work for my usecase.
Once again thanks a lot :)

**Comment by danthe3rd** [comment]
So I gave it a shot and it's working great! Thanks for putting this together :)
There is just one issue when `problem_size_0_n > 32 && problem_size_0_n % 2 == 1` (for instance `problem_size_0_n = 33`): it triggers an "CUDA Exception: Warp Misaligned Address" upon reading the shared-memory because it's not aligned on 32bits after the initial residual tile. I'm not sure if this is a specific case you want to support tho.

File: include/cutlass/gemm/warp/mma_tensor_op_tile_access_iterator.h [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+            MatrixCoord offset(
+              access_m_idx * 8 + inst_m_idx * InstructionShape::kRow, 
+              inner_idx * 4 * kElementsPerAccess + iterations_ * InstructionShape::kColumn);
+
+            MatrixCoord access_coord = origin_ + offset;
+
+//            if(access_coord.row() < extent_.row() && access_coord.column() < extent_.column()) {
+
+              access_ptr[access_idx] = *reinterpret_cast<AccessType const *>(
+                ref_.data() + ref_.offset(offset));
```
Comment (danthe3rd) [inline include/cutlass/gemm/warp/mma_tensor_op_tile_access_iterator.h:312]: support `problem_size_0_n = 33` would require to change this to 2 accesses, but will also be slower (maybe via template parameter?)

**Comment by hwu36** [comment]
We have no itention to support small alignment in b2b gemm.  It is not a good idea to apply b2b on not well aligned. Inputs.  Padding matices is easy and can immediately bring good performance.

**Review by kerrmudgeon — APPROVED** [review]

**Pull request merged**